integrations/crewai/README.md

```diff
@@ -0,0 +1,204 @@
+RETURNS STRING
+LANGUAGE PYTHON
+COMMENT 'Executes Python code and returns its stdout.'
+AS $$
+    import sys
+    from io import StringIO
+    stdout = StringIO()
+    sys.stdout = stdout
+    exec(code)
+    return stdout.getvalue()
+$$
+"""
+
+client.create_function(sql_function_body=sql_body)
+```
+
+Now that the function exists within the Catalog and Schema that we defined, we can interface with it from CrewAI using the ucai_crewai package.
+
+#### Create an instance of a CrewAI compatible tool
+
+[CrewAI Tools](https://docs.crewai.com/core-concepts/Tools/) are callable external functions that GenAI applications (called by
+an LLM), which are exposed with a UC interface through the use of the ucai_crewai package via the `UCFunctionToolkit` API.
+
+```python
+from ucai_crewai.toolkit import UCFunctionToolkit
+
+# Pass the UC function name that we created to the constructor
+toolkit = UCFunctionToolkit(function_names=[func_name])
+
+# Get the CrewAI-compatible tools definitions
+tools = toolkit.tools
+```
+
+If you would like to validate that your tool is functional prior to integrating it with CrewAI, you can call the tool directly:
+
+```python
+my_tool = tools[0]
+
+my_tool.fn(**{"code": "print(1)"})
+```
+
+#### Utilize our function as a tool within a CrewAI `Crew`
+
+With our interface to our UC function defined as a CrewAI tool collection, we can directly use it within a CrewAI `Crew`. 
+
+```python
+import os
+from crewai import Agent, Task, Crew
+
+# Set up API keys
+os.environ["OPENAI_API_KEY"] = "your key"
+
+# Create agents
+coder = Agent(
+    role="Simple coder",
+    goal= "Create a program that prints Hello Unity Catalog!",
+    backstory="likes long walks on the beach",
+    expected_output="string",
+    tools=tools,
+    verbose=True
+)
+
+reviewer = Agent(
+    role="reviewer",
+    goal="Ensure the researcher calls a function and shows the answer",
+    backstory="allergic to cats",
+    expected_output="string",
+    verbose=True
+)
+
+# Define tasks
+research = Task(
+    description="Call a tool",
+    expected_output="string",
+    agent=coder
+)
+
+review = Task(
+    description="Review the tool call output. Once complete, stop.",
+    expected_output="string",
+    agent=reviewer,
+)
+
+# Assemble a crew with planning enabled
+crew = Crew(
+    agents=[coder, reviewer],
+    tasks=[research, review],
+    verbose=True,
+    planning=True,  # Enable planning feature
+)
+
+# Execute tasks
+crew.kickoff()
+```
+
+Output
+```text
+[2024-10-08 14:29:25][INFO]: Planning the crew execution
+
+# Agent: Simple coder
+## Task: Call a tool1. **Identify the Objective**: The Simple Coder aims to create a program that prints "Hello Unity Catalog!" by utilizing the UnityCatalogTool.
+
+2. **Preparation for Tool Call**:
+   - Ensure that the Python environment is set and accessible to execute the code.
+   - Prepare the code snippet that needs to be executed. In this case, the Python code is as follows:
+     ```python
```

Comment:
This will render really oddly in markdown. This will probably need some escape indicators